File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans=-(1<<31)
        preSum=0
        for i in range(0,len(nums)):
            preSum=preSum+nums[i]
            ans=max(ans,preSum)
            if preSum<0:
                preSum=0
        return ans
        

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    
    let preSum=0
    let ans=-3246826389692868
    for(let i=0;i<nums.length;i++){
        preSum=preSum+nums[i];
        ans=Math.max(ans,preSum)
        if (preSum<0){
            preSum=0
        }
    }
    return ans
};